### PR TITLE
Trader rowtrader

### PR DESCRIPTION
**Pull request by hamidjackal**

### CODE REVIEW

**Comment by vd1371** [comment]
Thank you Hamid for your effort. Please consider the following suggestions:

1- To be consistent throughout the whole project, please re-write your example and use FXDataLoader. You may find an example in AroonHorn_FT.py on then master branch. This change will cause a problem in your main iteration. idx is now iterating over integers from 0 to len(df)-1. But when you change it, the index will be datetime. In that case, I suggest using for i, j in enumerate(ls).

2- Feature generator and decision model will be passed to the simulate or the __init__ of the class. This is my mistake. Please choose one of them based on our judgment. fg and dm must be passed to the class or the simulate method.

3- I know it's a pain in the ass, but please consider adding more comments to the code. It is really hard to understand your thoughts from some ifs. It will be helpful to everyone including ourselves in the future.

4- Line 62: I know row.atr is a perfect way to get a value from the panda data frame, but I can remember that when it comes to more complex analytics, such as multiprocessing, this way of addressing a value causes problems. It is safer to get values like atr['row']. Please apply this one in your code.

5- The decision made by the decision model is -1, 0, 1 corresponding to SHORT, WAIT, LONG

5- Current indexing at line 70 is inefficient. You could simply use df_aux.loc[opening_date, OPEN] or sth like this.

6- Lines 72 to 80  could be summarized and written in 4 lines with one less if analysis. For example, your decision is SHORT which is actually -1. Then, line 73 and 78 could be written as one single line:

pelle_price = main_opening_price + decision * atr_value

another example:
stop_loss = main_opening_price - decision * (1.5 * atr_value)

7- trade_status does not describe the content of it. I recommend trade_type as it also has been used in the Trade.py class. Please also revise it.

8- The final trailing stop loss that we reached together is a little bit different from what you have implemented. Let's discuss it in Telegram.

9- The purpose of writing Trade.py was to make traders look more beautiful and readable.
Please use this class in this file. Please look at my example for a possible substitution of lines 65 to 80:

if not trade_on and not trade_type:
            	trade_on = True
            	new_trade = Trade(self.main_pair_name, self.aux_pair_name)
            	new_trade.set_opening_info(opening_time = row['Date'],
            								main_open_price = row[OPEN],
            								aux_open_price = df_aux.loc[date, OPEN],
            								risk = 1.5,
            								balance = previous_balance)
            	pelle_price = row[OPEN] + trade_type * atr_value
            	stop_loss = row[OPEN] - trade_type * (1.5 * atr_value)

It has not considered the spread but I'm sure you got the idea.

10- When each trade is opened, we should know the balance, and we should have the balance at the end of the trade. The Trade.py module does it for you automatically. Please also remember this one.


This is all for now. I'm sure we will need a lot of discussion in the future to finalize it.
Many thanks in advance.
